bftengine/src/bftengine/ControlStateManager.cpp

```diff
@@ -34,13 +35,15 @@ void ControlStateManager::setStopAtNextCheckpoint(int64_t currentSeqNum) {
 }
 
 std::optional<int64_t> ControlStateManager::getCheckpointToStopAt() {
+  if (!enabled_) return {};
   if (!state_transfer_->loadReservedPage(getUpdateReservedPageIndex(), sizeOfReservedPage_, scratchPage_.data())) {
     return {};
   }
   std::istringstream inStream;
   inStream.str(scratchPage_);
   controlStateMessages::StopAtNextCheckpointMessage msg;
   concord::serialize::Serializable::deserialize(inStream, msg);
+  if (msg.seqNumToStopAt_ < 0) return {};
```

Comment:
This is unexpected, correct? If so, we should at least log a warning.

Reply:
WE shouldn't assert as it's a client message that was received, and that would allow for malicious operators to crash the system. However, we should almost certainly log an error message and/or throw an exception if we can handle it upstream. My guess is that we have many places in this code base where we should throw exceptions, but we don't have any way to handle them. Logging a very loud error is probably the best short term fix.